src/cmd/services/m3comparator/main/querier.go

```diff
@@ -124,18 +109,47 @@ func (q *querier) FetchCompressed(
 	options *storage.FetchOptions,
 ) (m3.SeriesFetchResult, m3.Cleanup, error) {
 	var (
-		// TODO: take from config.
+		iters encoding.SeriesIterators
+		err   error
+	)
+
+	name := q.opts.tagOptions.MetricName()
+	for _, matcher := range query.TagMatchers {
+		if bytes.Equal(name, matcher.Name) {
+			iters = q.handler.getSeriesIterators(string(matcher.Value))
```

Comment:
nit: Do you want to break if you set `iters` to something?

Reply:
Good call